src/transformers/generation/utils.py

```diff
@@ -2024,17 +2031,39 @@ def contrastive_search(
                 items = []
                 # item is either the key or the value matrix
                 for item in layer:
-                    items.append(item.repeat_interleave(top_k, dim=0))
+                    if low_memory:
+                        items.append(item.repeat_interleave(1, dim=0))
+                    else:
+                        items.append(item.repeat_interleave(top_k, dim=0))
                 new_key_values.append(items)
             model_kwargs["past_key_values"] = new_key_values
 
-            # compute the candidate tokens by the language model and collects their hidden_states
-            next_model_inputs = self.prepare_inputs_for_generation(top_k_ids.view(-1, 1), **model_kwargs)
-            outputs = self(
-                **next_model_inputs, return_dict=True, output_hidden_states=True, output_attentions=output_attentions
-            )
-            next_past_key_values = self._extract_past_from_model_output(outputs, standardize_cache_format=True)
+            # if the used memory exceeds a threshold, do not batch
```

Comment:
```suggestion
```

The comment suggests automatic detection of memory usage :)

Reply:
Very perceptive! That indeed was the initial goal but identifying the memory threshold required turned out to be difficult, so we instead have the keyword argument approach